### ISSUE TITLE
[🚀 Feature]: Need a way to disable GUI on Selenium Grid hub

### ISSUE THREAD

**Issue by joshea2018** [SeleniumHQ/selenium#12725]
### Feature and motivation

We need a way to disable the GUI which loads on the Selenium Grid Hub server. We need to do this in a way that doesn't impact the ability to register nodes and send jobs from clients to the hub which are then distributed to nodes running on separate servers. The best option would be to do this by passing a parameter when loading the .jar file (we run java -jar xxxx.jar ...).

If parameters aren't an option at this time, need to determine whether we can modify the jar to disable the GUI as discussed above.

### Usage example

We would run Selenium host using a parameter to disable the GUI.

Example: java -jar selenium-server-4.12.1.jar hub --port 4444 **"--noUI"** (where by adding the **noUI** parameter prevents the UI from loading)

**Comment by baflQA** [comment]
Thumbsup actually. When I run the grid on CI, I don't need any UI, and my agent would be happy to use any resources consumed by serving UI.

**Comment by krmahadevan** [comment]
@joshea2018 - Just curious. How is the Grid UI impacting the hub functionality ? It would be good if you could please help add more context to this ask, because from what I know, the Grid UI is just a easier way of finding out what is going on with the grid currently and does not have anything to do with any of the grid's capabilities.

@baflQA - Even if the Grid UI (which is kind of like a mandatory route for the hub) exists, it doesn't get in any way of the hub in terms of resource utilisation because it's just another route (or a http servlet if we are to talk in the servlet world).
Atleast this is how I know the Grid as.

**Comment by joshea2018** [comment]
It's more a security issue. Having a website running in a secure environment means that all security controls must be enforced (such as secure headers, anti-CSRF tokens, etc. With limitations on how to address these types of security items, it would be beneficial to simply turn off the ability to browse to the UI unless it's needed. Even better would be to address all the security issues, but that's a separate discussion, not meant to be discussed here. If the GUI is not necessary for the functionality for Grid, it would be good to be able to disable it until needed.

**Comment by diemol** [comment]
I am happy to accept a PR for this. You would need to add a boolean option in the [`RouterFlags`](https://github.com/SeleniumHQ/selenium/blob/43eb1e5477c9ef30bf6e2097ef984dec3f25077f/java/src/org/openqa/selenium/grid/router/httpd/RouterFlags.java#L32) and [`RouterOptions`](https://github.com/SeleniumHQ/selenium/blob/43eb1e5477c9ef30bf6e2097ef984dec3f25077f/java/src/org/openqa/selenium/grid/router/httpd/RouterOptions.java#L22), and then use it when hooking the UI to the [`Hub`](https://github.com/SeleniumHQ/selenium/blob/53cca49c3b90045c73d4381684368105df1ca2be/java/src/org/openqa/selenium/grid/commands/Hub.java#L185), the [`Router`](https://github.com/SeleniumHQ/selenium/blob/43eb1e5477c9ef30bf6e2097ef984dec3f25077f/java/src/org/openqa/selenium/grid/router/httpd/RouterServer.java#L144), and [`Standalone`](https://github.com/SeleniumHQ/selenium/blob/53cca49c3b90045c73d4381684368105df1ca2be/java/src/org/openqa/selenium/grid/commands/Standalone.java#L190). Please add some tests as well.

**Comment by joshea2018** [comment]
@diemol Let me know if you need anything from me. I'm new to this process, so if you got what you need, great. If you need me to do something, I'll need to know what to do ;) Thanks!

**Comment by diemol** [comment]
I don't have the time to work on this, but I left instructions above on how to do it in case someone has the time to do it.

**Comment by debanjanc01** [comment]
@diemol is this up for grabs?

**Comment by titusfortner** [comment]
Yes it's available.

**Comment by joshea2018** [comment]
Checking to see if anyone has made any progress on this request. We need a way to disable the GUI while maintaining the functionality of Selenium Grid. Loading a graphic interface isn't necessary for us except potentially for troubleshooting, so we need the ability to disable the GUI to mitigate vulnerabilities tied to the current implementation of the GUI. Please let me know if there are any updates. Thank you!

**Comment by diemol** [comment]
We have not had time to work on this. This is why I gave pointers on how you @joshea2018, or someone on your team could contribute to this.

**Comment by dev-velo** [comment]
Threw a pr up from my interpretation for the steps @diemol outlined but no clue on how to add a test for this to check if my changes were correct

**Comment by diemol** [comment]
Closed via #13212, thanks @RevealOscar for starting this.

**Comment by joshea2018** [comment]
@RevealOscar @diemol Thank you for including this in the release. I have a follow up question. I downloaded 4.16.1 and tried **_running java -jar selenium-server-4.16.1.jar hub ?_** but didn't see any help regarding the ability to disable. I'm hoping to get this release into our environment soon, but need to make sure I can disable the GUI in order to mitigate security findings. 

I did see #13212 as one of the commits listed, but don't see the code nor reference in the changelog for 4.16.1. If this isn't in the current release, I can wait. 

Sorry if I'm slow on responding here. I'm not a nature fit to GitHub. I'm more an infrastructure person. Let me know if you have any insight. Thanks!
